src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCaseEquivalences.cs

```diff
@@ -0,0 +1,165 @@
+        /// </summary>
+        /// <param name="c">The character being analyzed</param>
+        /// <param name="culture">The <see cref="CultureInfo"/> to be used to determine the equivalences.</param>
+        /// <param name="equivalences">If <paramref name="c"/> is involved in case conversion, then equivalences will contain the
+        /// span of character which should be considered equal to <paramref name="c"/> in a case-insensitive comparison.</param>
+        /// <returns><see langword="true"/> if <paramref name="c"/> is involved in case conversion; otherwise, <see langword="false"/></returns>
+        public static bool TryFindCaseEquivalencesForCharWithIBehavior(char c, CultureInfo culture, out ReadOnlySpan<char> equivalences)
+        {
+            if ((c | 0x20) == 'i' || (c | 0x01) == '\u0131')
+            {
+                RegexCaseBehavior mappingBehavior = GetRegexBehavior(culture);
+                equivalences = c switch
+                {
+                    // Invariant mappings
+                    'i' or 'I' when mappingBehavior is RegexCaseBehavior.Invariant => new[] { 'I', 'i' },
+
+                    // Non-Turkish mappings
+                    'i' when mappingBehavior is RegexCaseBehavior.NonTurkish => new[] { 'I', 'i', '\u0130' },
+                    'I' when mappingBehavior is RegexCaseBehavior.NonTurkish => new[] { 'I', 'i', },
+                    '\u0130' when mappingBehavior is RegexCaseBehavior.NonTurkish => new[] { 'i', '\u0130' },
+
+                    // Turkish mappings
+                    'I' or '\u0131' when mappingBehavior is RegexCaseBehavior.Turkish => new[] { 'I', '\u0131' },
+                    'i' or '\u0130' when mappingBehavior is RegexCaseBehavior.Turkish => new[] { 'i', '\u0130' },
+
+                    // Default
+                    _ => null
+                };
+                return equivalences != null;
+            }
+            else
+            {
+                return TryFindCaseEquivalencesForChar(c, out equivalences);
+            }
+        }
+
+        /// <summary>
+        /// Returns which <see cref="RegexCaseBehavior"/> should be used based on the passed in <paramref name="culture"/>.
+        /// </summary>
+        /// <param name="culture">The <see cref="CultureInfo"/> to be used to determine the behavior.</param>
+        /// <returns>The <see cref="RegexCaseBehavior"/> that should be used.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static RegexCaseBehavior GetRegexBehavior(CultureInfo culture)
+        {
+            if (IsCultureInvariant(culture.Name.AsSpan()))
+                return RegexCaseBehavior.Invariant;
+            else if (IsTurkishOrAzerbaycan(culture.Name.AsSpan()))
+                return RegexCaseBehavior.Turkish;
+            else
+                return RegexCaseBehavior.NonTurkish;
+
+            bool IsCultureInvariant(ReadOnlySpan<char> cultureName)
```

Comment:
>bool IsCultureInvariant(ReadOnlySpan<char> cultureName) [](http://example.com/codeflow?start=12&length=55)

This looks wrong to me passing the cultureName and not using it?

```C#

 bool IsCultureInvariant(ReadOnlySpan<char> cultureName) => cultureName.Length == 0;

```

Reply:
seems it can just be inlined.